app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,60 @@
+class RecipesController < ApplicationController
+  def index
+    @recipe = Recipe.includes(:user).all
+  end
+
+  def show
+    @recipe = Recipe.find(params[:id])
+    @foods = Food.all
+    @foods.each do |food|
+      @recipe_food = RecipeFood.find(food.id)
+    end
+  end
+
+  def shopping_list
+    @ids = RecipeFood.all.map(&:food_id)
+    puts @ids
+    @foods = Food.where(id: @ids)
+    @sum = 0
+    @foods.each do |food|
+      @recipe_food = RecipeFood.find_by(food_id: food.id)
+
+      puts @recipe_food.class
```

Comment:
- [ ] I would not recommend putting the `puts` statement in the controller. If anything you want to show in the UI please show through `view`.